typing/typedecl.ml

```diff
@@ -1850,6 +1854,7 @@ let update_decl_jkind env dpath decl =
     | Type_record (lbls, rep) ->
       let lbls, rep, type_jkind = update_record_kind decl.type_loc lbls rep in
       let type_jkind, type_has_illegal_crossings = add_crossings type_jkind in
+      let type_jkind = Jkind.mark_best type_jkind in
```

Comment:
I think there needs to be a `Note [Quality of jkinds during inference]` explaining the slightly-delicate dance going on in this file. It actually doesn't need to be a long Note -- not_best during inference, best for concrete types after inference -- but the choices here are subtle and the code is distributed. Having a Note will make it much easier for us to understand what's going on in 3 years.